### ISSUE TITLE
Expected Center Loss

### ISSUE THREAD

**Issue by KiryanovKD** [tensorflow/tensorflow#32379]
**System information**

TensorFlow version (you are using): 2 rc
Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
Expected Center Loss can help in learning discriminative features in seq training without framewising

**Will this change the current api? How?**
No

**Who will benefit with this feature?**
Anyone who wants to use this loss function in ASR or OCR

**Any Other info.**
https://arxiv.org/pdf/1905.06907.pdf

**Comment by dynamicwebpaige** [comment]
This feature request is outside of the scope of TensorFlow Core, but would be a welcome addition as a loss function in [TensorFlow Addons](https://www.github.com/tensorflow/addons). If expected center loss has considerable traction in addons, we will then graduate it to `tensorflow/tensorflow`.

Rerouting this ticket to `addons`; @seanpmorgan for visibility.
